src/extension.ts

```
@@ -0,0 +1,51 @@
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as preview from './preview';
+
+/**
+ * This method is called when the extension is activated.
+ * @param context The context in which the extension is activated.
+ */
+export function activate(context: vscode.ExtensionContext) {
+  const disposable = vscode.workspace.onDidOpenTextDocument(async document => {
+    if (isExecutable(document.fileName)) {
+      await preview.previewOutput(document.fileName);
+    }
+  });
+
+  context.subscriptions.push(disposable);
+
+  preview.activate(context);
+}
+
+/**
+ * This method is called when the extension is deactivated.
+ */
+export function deactivate() {
+  preview.deactivate();
+}
+
+/**
+ * Checks if a file is a binary executable.
+ * @param filePath The path to the file.
+ * @returns True if the file is a binary executable, false otherwise.
+ */
+function isExecutable(filePath: string): boolean {
+  if (!fs.existsSync(filePath)) {
+    return false;
+  }
+
+  try {
+    const stats = fs.statSync(filePath);
+
+    const isExecutable = (stats.mode & 0o111) !== 0;
+    const isWinExecutable =
+      filePath.endsWith('.exe') && os.platform() === 'win32';
+
+    return isExecutable || isWinExecutable;
+  } catch (error) {
+    console.error(error);
```

Avoid `console` calls. They are not helpful for users and mask the errors. I think removing `catch` and allowing the exception escape will be fine here. We can better decide how to handle that later.

---

I can add a new output channel in vscode, to log the errors, and any additional data which could be useful for issues reporting, and debugging

---

Sure. Do what you believe makes sense.